File: wrappers/mvc/demos/Kendo.Mvc.Examples/Controllers/Waterfall_Charts/Remote_Data_BindingController.cs
using System;
using System.Linq;
using System.Web.Mvc;
using Kendo.Mvc.Examples.Models;

namespace Kendo.Mvc.Examples.Controllers
{
    public partial class Waterfall_ChartsController : Controller
    {
        public ActionResult Remote_Data_Binding()
        {
            return View();
        }

        [HttpPost]
        public ActionResult _PriceData()
        {
            var data = new PriceData[] {
                new PriceData
                {
                    Name = "List",
                    Value = 100
                },
                new PriceData
                {
                    Name = "List\nDiscount",
                    Value = -5
                },
                new PriceData
                {
                    Name = "Invoice",
                    Summary = "total"
                },
                new PriceData
                {
                    Name = "Invoice\nDiscount",
                    Value = -6
                },
                new PriceData
                {
                    Name = "Rebates",
                    Value = -3
                },
                new PriceData
                {
                    Name = "Errors",
                    Value = -1.10m
                },
                new PriceData
                {
                    Name = "Pocket\nPrice",
                    Summary = "total"
                },
                new PriceData
                {
                    Name = "Cost",
                    Value = -57.10m
                },
                new PriceData
                {
                    Name = "Handling",
                    Value = -0.50m
                },
                new PriceData
                {
                    Name = "Pocket\nMargin",
                    Summary = "total"
                }
            };

            return Json(data);
        }
    }
}

File: wrappers/java/spring-demos/src/main/java/com/kendoui/spring/controllers/waterfallcharts/RemoteDataBindingController.java
package com.kendoui.spring.controllers.waterfallcharts;

import java.util.ArrayList;
import java.util.List;

import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestMethod;
import org.springframework.web.bind.annotation.ResponseBody;

import com.kendoui.spring.models.CashFlowData;
import com.kendoui.spring.models.ChartDataRepository;
import com.kendoui.spring.models.ElectricityProduction;
import com.kendoui.spring.models.PriceData;

@Controller("dataviz-waterfall_charts_remote_data-controller")
@RequestMapping(value="/waterfall-charts/")
public class RemoteDataBindingController {
    @RequestMapping(value = "/remote-data-binding", method = RequestMethod.GET)
    public String index() {
        return "/waterfall-charts/remote-data-binding";
    }
    
    @RequestMapping(value = "/remote-data-binding/read", method = RequestMethod.POST)
    public @ResponseBody List<PriceData> read() {
        List<PriceData> data = new ArrayList<PriceData>();
        data.add(new PriceData("List", 100));
        data.add(new PriceData("List\nDiscount", -5));
        data.add(new PriceData("Invoice", "total"));
        data.add(new PriceData("Invoice\nDiscount", -6));
        data.add(new PriceData("Rebates", -3));
        data.add(new PriceData("Errors", -1.1));
        data.add(new PriceData("Pocket\nPrice", "total"));
        data.add(new PriceData("Cost", -57.1));
        data.add(new PriceData("Handling", -0.5));
        data.add(new PriceData("Pocket\nMargin", "total"));
        
        return data;
    }
}
